radix-engine-profiling/src/rocks_db_metrics/mod.rs

```diff
@@ -92,513 +121,96 @@ impl<S: SubstateDatabase + CommittableSubstateDatabase> CommittableSubstateDatab
+        let db_update = &partition_update[0];
+        match db_update {
+            DatabaseUpdate::Set(value) => {
+                let exists = self.commit_set_metrics.borrow().get(&value.len()).is_some();
+                if exists {
+                    self.commit_set_metrics
+                        .borrow_mut()
+                        .get_mut(&value.len())
+                        .unwrap()
+                        .push(duration / database_updates.len() as u32);
+                } else {
+                    self.commit_set_metrics
+                        .borrow_mut()
+                        .insert(value.len(), vec![duration / database_updates.len() as u32]);
+                }
             }
-            println!(
-                "\rRound {}/{}  read time: {} s, left: {} s\r",
-                i + 1,
-                READ_REPEATS,
-                duration,
-                (READ_REPEATS - (i + 1)) * duration as usize
-            );
-        }
-
-        println!("Read done");
-    }
-
-    fn run_read_not_found_test<S: SubstateDatabase + CommittableSubstateDatabase>(
-        substate_db: &mut S,
-    ) {
-        println!("Read not found test start...");
-
-        let mut data_index_vector_2: Vec<(DbPartitionKey, DbSortKey)> = Vec::new();
-        let mut rng = rand::thread_rng();
-
-        for _ in 0..COUNT {
-            let mut node_id_value = [0u8; NodeId::UUID_LENGTH];
-            rng.fill(&mut node_id_value);
-            let node_id = NodeId::new(EntityType::InternalKeyValueStore as u8, &node_id_value);
-            let partition_key =
-                SpreadPrefixKeyMapper::to_db_partition_key(&node_id, PartitionNumber(0u8));
-
-            let mut substate_key_value = [0u8; NodeId::LENGTH];
-            rng.fill(&mut substate_key_value);
-            let sort_key =
-                SpreadPrefixKeyMapper::to_db_sort_key(&SubstateKey::Map(substate_key_value.into()));
-
-            data_index_vector_2.push((partition_key.clone(), sort_key));
-        }
-
-        for _ in 0..READ_REPEATS {
-            for (p, s) in data_index_vector_2.iter() {
-                let read_value = substate_db.get_substate(&p, &s);
-                assert!(read_value.is_none());
+            DatabaseUpdate::Delete => {
+                let exists = self
+                    .commit_delete_metrics
+                    .borrow()
+                    .get(&delete_value_len)
+                    .is_some();
+                if exists {
+                    self.commit_delete_metrics
+                        .borrow_mut()
+                        .get_mut(&delete_value_len)
+                        .unwrap()
+                        .push(duration / database_updates.len() as u32);
+                } else {
+                    self.commit_delete_metrics.borrow_mut().insert(
+                        delete_value_len,
+                        vec![duration / database_updates.len() as u32],
+                    );
+                }
```

Comment:
Minor: can be `.entry().or_default().push()`

Reply:
Nice tip!